pkg/operator/deploy/deploy.go

```diff
@@ -374,53 +374,68 @@ func (o *operator) IsReady(ctx context.Context) (bool, error) {
 	return true, nil
 }
 
-func checkOperatorDeploymentVersion(ctx context.Context, cli appsv1client.DeploymentInterface, name string, gitCommit string) error {
+func checkOperatorDeploymentVersion(ctx context.Context, cli appsv1client.DeploymentInterface, name string, desiredVersion string) (bool, error) {
 	d, err := cli.Get(ctx, name, metav1.GetOptions{})
-	if err != nil {
-		return err
+	switch {
+	case kerrors.IsNotFound(err):
+		return false, nil
+	case err != nil:
+		return false, err
 	}
-
-	if d.Labels["version"] != gitCommit {
-		return errors.New(name + " is not running the desired version: " + gitCommit)
+	if d.Labels["version"] != desiredVersion {
+		return false, nil
 	}
-
-	return nil
+	return true, nil
 }
 
-func checkPodImageVersion(ctx context.Context, cli corev1client.PodInterface, namespace string, gitCommit string) error {
-	podList, err := cli.List(ctx, metav1.ListOptions{LabelSelector: "app=" + namespace})
-	if err != nil {
-		return err
+func checkPodImageVersion(ctx context.Context, cli corev1client.PodInterface, role string, desiredVersion string) (bool, error) {
+	podList, err := cli.List(ctx, metav1.ListOptions{LabelSelector: "app=" + role})
+	switch {
+	case kerrors.IsNotFound(err):
+		return false, nil
+	case err != nil:
+		return false, err
 	}
 	for _, pod := range podList.Items {
 		imageTag := strings.Split(pod.Spec.Containers[0].Image, ":")
-		if imageTag[len(imageTag)-1] != gitCommit {
-			return errors.New(pod.Name + " pod of namespace " + pod.Namespace + " is not running the desired version: " + gitCommit)
+		if imageTag[len(imageTag)-1] != desiredVersion {
+			return false, nil
 		}
 	}
-	return nil
+	return true, nil
 }
 
-func (o *operator) IsRunningDesiredVersion(ctx context.Context) error {
-	// check if aro-operator-master is running desired version
-	err := checkOperatorDeploymentVersion(ctx, o.kubernetescli.AppsV1().Deployments(pkgoperator.Namespace), "aro-operator-master", version.GitCommit)
-	if err != nil {
-		return err
+func (o *operator) IsRunningDesiredVersion(ctx context.Context) (bool, error) {
+	// Get the desired Version
+	image := o.env.AROOperatorImage()
+	desiredVersion := version.GitCommit
```

Comment:
Let's not tie the `desiredVersion` to the `version.GitCommit`.  In local dev mode it will fail always.  Instead do something like this

```
	desiredVersion := "latest"
	if strings.Contains(image, ":") {
		str := strings.Split(image, ":")
		desiredVersion = str[len(str)-1]
	}